subprojects/build-events/src/main/java/org/gradle/internal/build/event/types/DefaultCompositeTestSource.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2025 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.internal.build.event.types;
+
+import org.gradle.tooling.internal.protocol.events.InternalCompositeTestSource;
+import org.gradle.tooling.internal.protocol.events.InternalTestSource;
+import org.jspecify.annotations.NullMarked;
+
+import java.io.Serializable;
+import java.util.List;
+
+@NullMarked
+public class DefaultCompositeTestSource implements InternalCompositeTestSource, Serializable {
```

Comment:
🤔 Could we move all these source types to a new sub-package, and add a `package-info` file explaining that the hierarchy here is meant to mirror the hierarchy of `org.junit.platform.engine.TestSource`?